### PR TITLE
Add Queen Attack exercise

### PR DESCRIPTION
**Pull request by CRivasGomez**
Hey!
I added `queen-attack` exercise 🙂

### CODE REVIEW

File: exercises/queen-attack/queen-attack.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import Queens from './queen-attack'
```
Comment (masters3d) [inline exercises/queen-attack/queen-attack.test.ts]: This should be call QueenAttack

File: exercises/queen-attack/queen-attack.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+describe('Queens', () => {
+    it('has the correct default positions', () => {
+        const queens = new Queens()
+        expect(queens.white).toEqual([0, 3])
```
Comment (masters3d) [inline exercises/queen-attack/queen-attack.test.ts]: I don't think we should expose this. I'd rather see an error being thrown.

Comment (masters3d) [reply exercises/queen-attack/queen-attack.test.ts]: as in, it white should probably be null

File: exercises/queen-attack/queen-attack.test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            '_ _ _ _ _ _ _ _',
+            '_ _ _ _ _ _ _ _',
+            '_ _ _ _ _ _ _ _',
+            '_ _ _ _ _ _ B _',
+            '_ _ _ _ _ _ _ _\n'].join('\n')
+        expect(queens.toString()).toEqual(board)
+    })
+
+    xit('queens cannot attack', () => {
+        const queens = new Queens({ white: [2, 3], black: [4, 7] })
```
Comment (masters3d) [inline exercises/queen-attack/queen-attack.test.ts]: What happens if somebody provides an array that is too short  or too long?

**Comment by masters3d** [comment]
I actually think this can be better models as

```js
let q1 = new Queen(position1)
let q2 = new Queen(position2)
q1.canAttack(q2)

//If you wanted to expand it then you can also have a board

let b8by8 = new Board(8, 8)
b8by8.add(q1)
b8by8.add(q2)
b8by8.draw() // Same as two string
```

File: exercises/queen-attack/queen-attack.test.ts (hunk 2 above)
Comment (CRivasGomez) [reply exercises/queen-attack/queen-attack.test.ts]: I just enforced the necessity of using a parameter for queens initial position.

File: exercises/queen-attack/queen-attack.test.ts (hunk 3 above)
Comment (CRivasGomez) [reply exercises/queen-attack/queen-attack.test.ts]: I just enforced them to be arrays of two `number` elements.

**Pull request merged**